packages/pirania/files/usr/lib/lua/voucher/vouchera.lua

```diff
@@ -40,21 +41,21 @@ function voucher_init(obj)
     end
     voucher.mac = obj.mac
 
-    if obj.expiration_date == nil and obj.duration_m == nil then
-        return nil
-    elseif obj.expiration_date ~= nil then
-        voucher.expiration_date = obj.expiration_date
-    elseif obj.duration_m ~= nil then
-        voucher.duration_m = obj.duration_m
-    end
+    voucher.duration_m = obj.duration_m -- use nil to create a permanent voucher
+
+    voucher.expiration_date = obj.expiration_date
+
+    voucher.creation_date = obj.creation_date or os.time()
```

Comment:
In order to implement expiration_date as a derived property.
As this function is used to load the database from the json file, one must ensure that all entries have their creation_date already set.

Reply:
Oh, sorry I dragged the error from a previous comment. What I said has no sense.
What I do say now is:
Setting default values for unspecified properties like in `obj.creation_date or os.time()` makes sense when creating a voucher, but does not when loading a voucher from the database.
We could move `creation_date = os.time()` to voucher.add
And ignoring, with an error message, those vouchers passed to voucher_init without a creation date.
What do you think?

Reply:
You are thinking it like a general software upgrade issue? (where we need to support vouchers that are in already stablished databases and now need to have a creation date because we are changing the database schema?)
Or you think it as a general progamming "good practices" ?

Now this leaves me thinking that we may want to add a "version" attribute (maybe major/minnor) to the vouchers so we can upgrade the schema when vouchers with different major versions are discovered and maybe dont load vouchers of unsupported major versions?

Reply:
I was thinking on "good practices" only, but I think it's great to plan beforehand the software upgrade issue.
I think we can add it for sure! v0.1 ? :)

Reply:
i've done what you suggested! I think is better now, thanks.